cylc/flow/suite_files.py

```diff
@@ -722,3 +779,105 @@ def _locate_item(item, path):
     fname = os.path.join(path, item)
     if os.path.exists(fname):
         return fname
+
+
+def return_key_locations(store_dir):
+    """Return the paths to a directory's key files: (public, private)."""
+    return (
+        os.path.join(store_dir, UserFiles.Auth.PUBLIC_KEY_DIRNAME),
+        os.path.join(store_dir, UserFiles.Auth.PRIVATE_KEY_DIRNAME)
+    )
+
+
+def lockdown_private_keys(key_file_path):
+    """Change key file permissions to lock from other users."""
+    # This means that the owner can read & write, but all others cannot do
+    # anything, to the file, i.e. 600 ('-rw-------.') Linux file permission.
+    if not os.path.exists(key_file_path):
+        raise FileNotFoundError(
+            "Private key not found at location '%s'." % key_file_path)
+    os.chmod(key_file_path, stat.S_IRUSR | stat.S_IWUSR)
+
+
+def generate_key_store(store_parent_dir, keys_tag):
+    """Generate two sub-directories, each having a file with a CURVE key.
+
+    WARNING: be careful testing this. It uses 'shutil.rmtree' which will
+    wipe the whole 'store_dir/UserFiles.Auth.DIRNAME' directory if it exists.
+
+    Note: store_parent_dir must already exist as a valid directory.
+    """
+
+    # Define the directory structure to store the CURVE keys in
+    store_dir = os.path.join(store_parent_dir, UserFiles.Auth.DIRNAME)
+    public_key_loc, private_key_loc = return_key_locations(store_dir)
+
+    # Create, or wipe, that directory structure
+    for directory in [store_dir, public_key_loc, private_key_loc]:
+        if os.path.exists(directory):
+            shutil.rmtree(directory)  # WARNING: destructive, so take care
+        os.mkdir(directory)
+
+    # Make a new public-private CURVE key pair
+    zmq.auth.create_certificates(store_dir, keys_tag)
+
+    # Move key pair to appropriate directories & lock private key file
+    for key_file in os.listdir(store_dir):
+        if key_file.endswith(UserFiles.Auth.PUBLIC_TAG):
+            shutil.move(os.path.join(store_dir, key_file),
+                        os.path.join(public_key_loc, '.'))
+            # The public key keeps standard 644 ('-rw-r--r--') file permission
+        elif key_file.endswith(UserFiles.Auth.PRIVATE_TAG):
+            loc = shutil.move(os.path.join(store_dir, key_file),
+                              os.path.join(private_key_loc, '.'))
+            # Now lock the private key in its permanent location
+            try:
+                lockdown_private_keys(loc)  # Linux File Permission now 600
```

Comment:
:disappointed:

Reply:
Done - `.cylc/auth` & `.service/auth` is now `0700` on creation.. I don't think we need to refuse action/service if auth directory permissions vary as:
- If permissions are checked on each client call (not solely workflow startup), then this would probably add more overhead to the comms.
- I don't think we need to be too opinionated; i.e. users should be able to liberate the auth key-store post creation if desired.